Wikipedia/Code/WMFRecentPagesDataSource.m

```diff
@@ -59,6 +59,26 @@ - (nonnull instancetype)initWithRecentPagesList:(MWKHistoryList*)recentPages sav
             [dataSource removeItemAtIndexPath:indexPath];
             [[NSNotificationCenter defaultCenter] addObserver:dataSource selector:@selector(rebuildSections) name:MWKHistoryListDidUpdateNotification object:recentPages];
         };
+
+        [self.KVOController observe:self.recentPages keyPath:WMF_SAFE_KEYPATH(self.recentPages, entries) options:NSKeyValueObservingOptionInitial | NSKeyValueObservingOptionPrior block:^(WMFRecentPagesDataSource* observer, MWKHistoryList* object, NSDictionary* change) {
+            BOOL isPrior = [change[NSKeyValueChangeNotificationIsPriorKey] boolValue];
+            NSKeyValueChange changeKind = [change[NSKeyValueChangeKindKey] unsignedIntegerValue];
+            NSIndexSet* indexes = change[NSKeyValueChangeIndexesKey];
+
+            if (isPrior) {
```

Comment:
would be nice to not have to forward these KVO calls, but if we can't think of a way now it's fine.